File: modules/tv-recommendations/android/src/main/java/expo/modules/tvrecommendations/TvRecommendationsModule.kt
package expo.modules.tvrecommendations

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class TvRecommendationsModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("TvRecommendations")

    Function("syncRecommendations") { json: String ->
      val context = appContext.reactContext ?: return@Function false
      TvRecommendationsPublisher.sync(context, json)
    }

    Function("clearRecommendations") {
      val context = appContext.reactContext ?: return@Function false
      TvRecommendationsPublisher.clear(context)
    }

    Function("refreshRecommendations") {
      val context = appContext.reactContext ?: return@Function false
      TvRecommendationsPublisher.refreshFromCache(context)
    }
  }
}

File: modules/tv-recommendations/src/TvRecommendationsModule.ts
import { requireNativeModule } from "expo-modules-core";
import { Platform } from "react-native";
import type { TvRecommendationsModuleType } from "./TvRecommendations.types";

let TvRecommendationsModule: TvRecommendationsModuleType | null = null;

if (Platform.OS === "android" && Platform.isTV) {
  try {
    TvRecommendationsModule =
      requireNativeModule<TvRecommendationsModuleType>("TvRecommendations");
  } catch {
    TvRecommendationsModule = null;
  }
}

export function syncTvRecommendations(json: string): boolean {
  return TvRecommendationsModule?.syncRecommendations(json) ?? false;
}

export function clearTvRecommendations(): boolean {
  return TvRecommendationsModule?.clearRecommendations() ?? false;
}

export function refreshTvRecommendations(): boolean {
  return TvRecommendationsModule?.refreshRecommendations() ?? false;
}
